src/main/webapp/app/account/register/register.component.ts

```diff
@@ -61,10 +68,17 @@ export class RegisterComponent implements AfterViewInit {
     } else {
       const login = this.registerForm.get(['login'])!.value;
       const email = this.registerForm.get(['email'])!.value;
-      this.registerService.save({ login, email, password, langKey: this.languageService.getCurrentLanguage() }).subscribe(
-        () => (this.success = true),
-        response => this.processError(response)
-      );
+      this.registerService
+        .save({ login, email, password, langKey: this.languageService.getCurrentLanguage() })
+        .pipe(
+          switchMap(() =>
+            this.invitationToken ? this.invitationService.assignUserByLoginToInvitation(login, this.invitationToken) : of(null)
```

Comment:
ja um ehrlich zu sein habe ich mich mit dem of(null) zu schnell zu frieden gegeben. aber im subscribe mache ich das nicht, weil man sonst nested subscriptions hat

Reply:
aber bin da nicht der experte. https://blog.angular-university.io/rxjs-error-handling/ vllt mal hier mehr nachlesen (ich inklusive)

Reply:
> aber im subscribe mache ich das nicht, weil man sonst nested subscriptions hat

achso, das hatte ich nicht bedacht. hm ja, nur nicht dass er dann im fehler Fall was zu posten versucht.

Reply:
de link isch sick! gar ned gwüsst wases alles git. i dem fall weri elwe es "Catch and Rethrow" mittels `catchError` abracht. was meinsch?

Reply:
Verstehe einfach nicht genau, was der Vorteil davon ist. Wenn ja das Registrieren des users failt geht er im stream nicht weiter und fällt sofort in den error Fall. Wieso jetzt hier plötzlich die error abfangen und irgendwas machen? Viel besser wäre eben ein intercepter, der die http errors abfängt (was es ja schon gibt. Sonst würden ja die roten Balken mit den server errors gar nicht angezeigt werden).

Reply:
das war mir jetzt vom Blog Eintrag her nicht ersichtlich, dass `pipe` dann nicht ausgeführt wird xD aber ja, hab's überprüft. funktioniert so 👍🏼

Reply:
Ja es ist sicher nicht ein einfach Thema aber:

> In order to understand error handling in RxJs, we need to first understand that any given stream can only error out once. This is defined by the Observable contract, which says that a stream can emit zero or more values.
> 
> The contract works that way because that is just how all the streams that we observe in our runtime work in practice. Network requests can fail, for example.
> 
> A stream can also complete, which means that:
> 
> the stream has ended its lifecycle without any error
> after completion, the stream will not emit any further values
> As an alternative to completion, a stream can also error out, which means that:
> 
> the stream has ended its lifecycle with an error
> after the error is thrown, the stream will not emit any other values
> Notice that completion or error are mutually exclusive:
> 
> if the stream completes, it cannot error out afterwards
> if the streams errors out, it cannot complete afterwards
> Notice also that there is no obligation for the stream to complete or error out, those two possibilities are optional. But only one of those two can occur, not both.


Das sagt ja im Wesentlichen, dass wenn im stream ein error passiert, der stream nicht weitergeht und somit terminiert.